src/ChannelledSocket.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Amp\Sync;
+
+use Amp\ByteStream\ClosableStream;
+use Amp\ByteStream\ReadableResourceStream;
+use Amp\ByteStream\ResourceStream;
+use Amp\ByteStream\WritableResourceStream;
+use Amp\Cancellation;
+use Amp\Serialization\Serializer;
+
+/**
+ * Uses stream resources to create a Channel.
+ *
+ * @template TReceive
+ * @template TSend
+ * @template-implements Channel<TReceive, TSend>
+ */
+final class ChannelledSocket implements Channel, ClosableStream, ResourceStream
```

Comment:
Why does this class exist instead of using `ChannelledStream` directly?

Reply:
Prior, this class had a `close` method where as `ChanneledStream` did not because streams weren't always closable. Since this has changed I don't think this needs to exist.